src/domain/event/utils.ts

```diff
@@ -115,3 +121,66 @@ export const getEventPayload = (
     },
   };
 };
+
+export const getExistingVenuePayload = ({
+  venueData,
+  selectedLanguage,
+  formValues: { locationDescription, location: locationId },
+}: {
+  venueData: VenueQuery;
+  selectedLanguage: Language;
+  formValues: EventFormFields;
+}) => {
+  return {
+    venue: {
+      id: locationId,
+      // TODO: handle these 2 fields
+      hasClothingStorage: false,
+      hasSnackEatingPlace: false,
+      translations: [
+        ...(venueData?.venue?.translations
+          .map((t) => omit(t, ['__typename']))
+          .filter((t) => t.languageCode !== selectedLanguage.toUpperCase()) ||
+          []),
+        {
+          languageCode: selectedLanguage.toUpperCase() as TranslationLanguage,
+          description: locationDescription,
+        },
+      ],
+    },
+  };
+};
+
+export const getNewVenuePayload = ({
+  formValues,
+  selectedLanguage,
+}: {
+  formValues: EventFormFields;
+  selectedLanguage: Language;
+}) => {
+  return {
+    venue: {
+      id: formValues.location,
+      // TODO: handle these 2 fields
+      hasClothingStorage: false,
+      hasSnackEatingPlace: false,
+      translations: [
+        {
+          languageCode: selectedLanguage.toUpperCase() as TranslationLanguage,
+          description: formValues.locationDescription,
+        },
+      ],
+    },
+  };
+};
+
+export const getLocationDescription = (
+  eventData: EventQuery | null,
```

Comment:
If you are going to write helper function to get venue description you could use it also here